src/types/BatchClaims.sol

```diff
@@ -20,3 +20,13 @@ struct BatchClaim {
     string witnessTypestring; // Witness typestring appended to existing typestring.
     BatchClaimComponent[] claims; // The claim token IDs, recipients and amounts.
 }
+
+using BatchClaimsLib for BatchClaim global;
+
+library BatchClaimsLib {
+    function asRawPtr(BatchClaim calldata claim) internal pure returns (uint256 rawClaimPtr) {
```

Comment:
Two comments:
- I like how this simplifies things but I'm pretty sure it's not a zero-cost abstraction to do this as opposed to function casts since solidity will run independent validation on the inputs; if there's a way to keep some function casts while still simplifying the flows I think we should consider it
- we need natspec for these